packages/core/src/animation/Animator.ts

```diff
@@ -0,0 +1,465 @@
+  }
+
+  private _animatorController: AnimatorController;
+  private _diffValueFromBasePos: InterpolableValue;
+  private _diffFloatFromBasePos: number = 0;
+  private _diffVector2FromBasePos: Vector2 = new Vector2();
+  private _diffVector3FromBasePos: Vector3 = new Vector3();
+  private _diffVector4FromBasePos: Vector4 = new Vector4();
+  private _diffQuaternionFromBasePos: Quaternion = new Quaternion();
+  private _tempVector3: Vector3 = new Vector3();
+  private _tempQuaternion: Quaternion = new Quaternion();
+
+  /**
+   * @param entity - The entitiy which the animator component belongs to.
+   */
+  constructor(entity: Entity) {
+    super(entity);
+  }
+
+  /**
+   * Plays a state by name.
+   * @param stateName The state name.
+   * @param layerIndex The layer index(default 0).
+   * @param normalizedTime The time offset between 0 and 1(default 0).
+   */
+  playState(stateName: string, layerIndex: number = 0, normalizedTime: number = 0): void {
+    const { animatorController } = this;
+    if (!animatorController) return;
+    const animLayer = animatorController.layers[layerIndex];
+    const theState = animLayer.stateMachine.findStateByName(stateName);
+    theState.frameTime = theState.clip.length * normalizedTime;
+    animLayer._playingState = theState;
+    this.recorderMode = AnimatorRecorderMode.Playback;
+  }
+
+  /**
+   * Set the Animator in playback mode.
+   */
+  play(): void {
+    this.recorderMode = AnimatorRecorderMode.Playback;
+  }
+
+  /**
+   * Stops the animator playback mode.
+   */
+  stop(): void {
+    this.recorderMode = AnimatorRecorderMode.Offline;
+  }
+
+  /**
+   * Evaluates the animator component based on deltaTime.
+   * @param deltaTime - The deltaTime when the animation update.
+   * @private
+   */
+  update(deltaTime: number): void {
+    if (this.recorderMode !== AnimatorRecorderMode.Playback) return;
+    const { animatorController } = this;
+    if (!animatorController) return;
+    const { layers } = animatorController;
+    for (let i = 0; i < layers.length; i++) {
+      const isFirstLayer = i === 0;
+      const animLayer = layers[i];
+      if (!animLayer._playingState) {
+        animLayer._playingState = animLayer.stateMachine.states[0];
+      }
+      const currentState = animLayer._playingState;
+      currentState.frameTime += deltaTime / 1000;
+      if (currentState._playType === PlayType.IsFading) {
+        const fadingState = animLayer._fadingState;
+        if (fadingState) {
+          fadingState.frameTime += deltaTime / 1000;
+          if (fadingState.frameTime > fadingState.clipEndTime) {
+            fadingState.frameTime = fadingState.clipEndTime;
+          }
+        }
+      }
+      this._updatePlayingState(currentState, animLayer, isFirstLayer, deltaTime);
+    }
+  }
+
+  /**
+   * crossFade to the AnimationClip by name.
+   */
+  crossFade(
+    name: string,
+    layerIndex: number,
+    normalizedTransitionDuration: number,
+    normalizedTimeOffset: number
```

Comment:
I think `normalizedTimeOffset ` name  should same with `playState()` third param.

Reply:
done